pennylane/templates/subroutines/qdrift.py

```diff
@@ -178,7 +189,7 @@ def __init__(  # pylint: disable=too-many-arguments
                 "coefficients of the input Hamiltonian."
             )
 
-        if decomposition is None:  # need to do this to allow flatten and _unflatten
+        if decomposition is None:  # need to do this to allow flatten and unflatten
```

Comment:
@Jaybsoni Do you remember why flattening and unflattening required doing the decomposition on initialization?

Reply:
Ahh, I can't remember off the top of my head.

Reply:
Had a quick chat with @albi3ro offline. We concluded that it makes sense to move `_sample_decomposition` down to the `compute_decomposition` method and deprecate the `decomposition` kwarg. We can add a note to the doc-string instructing users to just extract the decomposition and use `qml.apply` to apply it instead of passing it to sub-sequent instances of QDrift.

Reply:
So we move `_sample_decomposition` into `compute_decomposition`.  

Then, to make `op.decomposition()` the same between calls, we set:
 ```
self.hyperparameters['seed'] = np.random.randint(0, high=10000000) if seed is None else seed
```

So the seed is randomly chosen on initialization, but fully set when we are doing decompositions. Randomness on initialization, consistency on decomposition.

Reply:
Ah okay! I now just skipped the `"decomposition"` kwarg in `_unflatten`, leading to a new sample (with same user-provided seed) when flattening and unflattening.
But I'm fine with the above as well.